src/jsonstream.cpp

```diff
@@ -0,0 +1,201 @@
+//
+// Created by jpiotte on 10/23/19.
+//
+
+#include <simdjson/isadetection.h>
+#include <atomic>
+#include <simdjson/padded_string.h>
+#include "simdjson/jsonstream.h"
+
+using namespace simdjson;
+
+//Architecture find_best_supported_implementation();
+//int json_parse_implementation(ParsedJson &pj, bool realloc_if_needed);
+//
+//
+//// function pointer type for json_parse
+//using json_parse_functype = int(const char *buf, size_t len, size_t batch_size, ParsedJson &pj, bool realloc_if_needed);
+//
+//// Pointer that holds the json_parse implementation corresponding to the
+//// available SIMD instruction set
+//extern std::atomic<json_parse_functype *> json_parse_ptr;
+//
+
+
+
+
+
+
+JsonStream::JsonStream(const char *buf, size_t len, size_t batch_size) {
```

Comment:
It's a minor thing, but should we not use the initializer list here instead of initializing in the constructor body?

Reply:
I think the current way is more versatile.  Say the user wants to use Memory Mapping and only has a buffer pointer and the size of the file, I don't think the initializer_list would work in this case.

Reply:
Sorry, maybe I was being unclear. I meant the concept of an initializer list (as opposed to e.g., `std::initializer_list`) meaning we'd write `JsonStream::JsonStream(const char* buf, size_t len, size_t batch_size) : buf_(buf), len_(len), batch_size_(batch_size) { }`.

Reply:
Oh.. I see.  Okay, sounds cool!

Thanks!